kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/kafkaservice/KafkaServiceReconciler.java

```diff
@@ -36,14 +48,26 @@ public UpdateControl<KafkaService> reconcile(KafkaService resource, Context<Kafk
                 .withObservedGeneration(resource.getMetadata().getGeneration())
                 .withStatus(Condition.Status.TRUE)
                 .build();
-        final KafkaService amended = resource
-                .edit()
-                .withNewStatus()
-                .withObservedGeneration(resource.getMetadata().getGeneration())
-                .withConditions(acceptedCondition)
-                .endStatus()
+        UpdateControl<KafkaService> uc = UpdateControl.patchStatus(newServiceWithCondition(resource, acceptedCondition));
+        LOGGER.info("Completed reconciliation of {}/{}", namespace(resource), name(resource));
+        return uc;
+    }
+
+    private static KafkaService newServiceWithCondition(KafkaService resource, Condition acceptedCondition) {
+        // @formatter:off
+        final KafkaService amended = new KafkaServiceBuilder()
```

Comment:
are we assigning to `amended` just so we can turn the formatter on and off? Or can it be inlined now its not being passed directly to `patchStatus`

Reply:
I can inline it.

Another thing I noticed is that this reconciler names the `KafkaService` parameter `resource`. The `KafkaProxyReconciler` calls it `primary`. The rest name the parameter after the type (so like `filter` or `ingress`). We should pick one and stick to it. I think I prefer naming it after the type because I don't really find the `primary` terminology particularly helpful, and all the things (primaries, secondaries and dependent) are resources.

Reply:
Naming after the type probably makes the most sense.